src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,226 @@
+
+    with self.argument_context('blueprint resource-group remove') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='A unique name of this resource group artifact.')
+
+    with self.argument_context('blueprint resource-group show') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='A unique name of this resource group artifact.')
+
+    with self.argument_context('blueprint resource-group list') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint artifact policy create') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+        c.argument('display_name', help='DisplayName of this artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('policy_definition_id', options_list=['--policy-definition'], help='The full policy definition id.')
+        c.argument('resource_group_art', help='Name of the resource group artifact to which the policy will be assigned.')
+        c.argument('parameters', arg_type=parameter_type, help='Parameters for policy assignment artifact. It can be a JSON string or JSON file path with "@" prefix.')
+
+    with self.argument_context('blueprint artifact policy update') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+        c.argument('display_name', help='DisplayName of this artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('resource_group_art', help='Name of the resource group artifact to which the policy will be assigned.')
+        c.argument('parameters', arg_type=parameter_type, help='Parameters for policy assignment artifact. It can be a JSON string or JSON file path with "@" prefix.')
+
+    with self.argument_context('blueprint artifact role create') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+        c.argument('display_name', help='DisplayName of this artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('role_definition_id', options_list=['--role-definition'], help='The full role definition id.')

```

Comment:
I have two more questions here:
1. Can we support custom role definition?
2. Why not use role name to specify a role definition?

Reply:
1. Blueprint does not support custom role definition.
2. The scope of blueprint and the scope of role are different, without knowing the scope of role, let's avoid using a simple role name although it seems now only roles in "/" scope are supported for blueprint.